src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -126,6 +149,17 @@ const DateTimeBlock : React.FC<IProps> = ({
   }, [date, displayMinutes, setDateCallback]);
 
   const setDateMinutes = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
+    const isContainsSymbolN =  /^[0-9]+[0-9]*$/;
+    if(value === '000'|| value.length>2){
+      return;
+    } else if (Number(value)>59){
+      value = '59';
+    } else if (!isContainsSymbolN.test(value)) {
+      if(value !== ''){
+        return ;
+      }
+    }
```

Comment:
I think this whole validation checks can be optimized with a regular expression.
**RegEx**:  `^[0-5][0-9]$`
Please try this and find if it fails or not to satisfy all the conditions.

Reply:
Done